app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -530,6 +529,9 @@ public StudyParticipant updateIdentifiers(Study study, CriteriaContext context,
             !study.isAutoVerificationEmailSuppressed()) {
             accountWorkflowService.sendEmailVerificationToken(study, account.getId(), account.getEmail());
         }
+        if (assignExternalId) {
+            externalIdService.assignExternalId(study, account.getExternalId(), account.getHealthCode());
+        }
```

Comment:
Should there be a test that verifies that not assigning an external ID doesn't try to assign the external ID again?

Reply:
Added, and added more verifications that the externalIdService is not called in scenarios where no assignment of the ext ID should be taking place.